mlflow/pmdarima.py

```diff
@@ -0,0 +1,364 @@
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.utils.file_utils import write_to
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+from mlflow.utils.environment import (
+    _mlflow_conda_env,
+    _validate_env_arguments,
+    _CONDA_ENV_FILE_NAME,
+    _process_pip_requirements,
+    _process_conda_env,
+    _CONSTRAINTS_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+)
+from mlflow.utils.docstring_utils import format_docstring, LOG_MODEL_PARAM_DOCS
+
+
+FLAVOR_NAME = "pmdarima"
+_MODEL_BINARY_KEY = "data"
+_MODEL_BINARY_FILE_NAME = "model.pmd"
+_MODEL_TYPE_KEY = "model_type"
+
+_logger = logging.getLogger(__name__)
+
+
+def get_default_pip_requirements():
+    """
+    :return: A list of default pip requirements for MLflow Models produced by this flavor.
+             Calls to :func:`save_model()` and :func:`log_model()` produce a pip environment that,
+             at a minimum, contains these requirements.
+    """
+    return [_get_pinned_requirement("pmdarima")]
+
+
+def get_default_conda_env():
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements())
+
+
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
+def save_model(
+    pmdarima_model,
+    path,
+    conda_env=None,
+    mlflow_model=None,
+    signature: ModelSignature = None,
+    input_example: ModelInputExample = None,
+    pip_requirements=None,
+    extra_pip_requirements=None,
+):
+    """
+    Save a pmdarima ``ARIMA`` model or ``Pipeline`` object to a path on the local file system.
+
+    :param pmdarima_model: pmdarima ``ARIMA`` or ``Pipeline`` model that has been ``fit`` on a
+                           temporal series.
+    :param path: Local path destination for the serialized model (in pickle format) is to be saved.
+    :param conda_env: {{ conda_env }}
+    :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+    :param signature: :py:class:`Model Signature <mlflow.models.ModelSignature>` describes model
+                      input and output :py:class:`Schema <mlflow.types.Schema>`. The model
+                      signature can be :py:func:`inferred <mlflow.models.infer_signature>`
+                      from datasets with valid model input (e.g. the training dataset with target
+                      column omitted) and valid model output (e.g. model predictions generated on
+                      the training dataset), for example:
+
+                      .. code-block:: py
+
+                      from mlflow.models.signature import infer_signature
+
+                      model = pmdarima.auto_arima(data)
+                      predictions = model.predict(n_periods=30, return_conf_int=False)
+                      signature = infer_signature(data, predictions)
```

Comment:
```suggestion
                      .. code-block:: py

                          from mlflow.models.signature import infer_signature

                          model = pmdarima.auto_arima(data)
                          predictions = model.predict(n_periods=30, return_conf_int=False)
                          signature = infer_signature(data, predictions)
```

Can we indent this block?

Reply:
yep!